packages/check-bundle/index.js

```diff
@@ -0,0 +1,20 @@
+// @ts-check
+import * as fs from 'fs';
+import * as crypto from 'crypto';
+import { checkBundle as checkBundleBytes } from './lite.js';
+
+/** @param {Uint8Array} bytes */
+const computeSha512 = bytes => {
+  const hash = crypto.createHash('sha512');
+  hash.update(bytes);
+  return hash.digest().toString('hex');
+};
+
+/**
+ * @param {string} path
+ * @param {string=} name
+ */
+export const checkBundle = async (path, name = undefined) => {
```

Comment:
Thanks for the batteries-included form. My use case has bytes in memory rather than a file on disk, so `controller.validateAndInstallBundle()` would need to define its own `computeSha512` and use `checkBundleBytes` from `lite.js`, in case that influences what API you want to give the non-lite version.

Reply:
I resolved this issue by adding a second method to the batteries-included interface.